tools/oeedger8r/Emitter.ml

```diff
@@ -1,190 +1,191 @@
-(* 
-  Copyright (c) Microsoft Corporation. All rights reserved.
-  Licensed under the MIT License. 
-*)
+(* Copyright (c) Microsoft Corporation. All rights reserved.
+   Licensed under the MIT License. *)
+
+(** This module is Open Enclave's plugin for Intel's Edger8r, allowing
+    us to share the same Enclave Definition Language, but emit our
+    SDK's bindings. *)
 
 open Ast
 open Plugin
 open Printf
-open Util  
+open Util
 
-(**************************** Begin Code borrowed and tweaked from CodeGen.ml *******************************************************)      
+(** ----- Begin code borrowed and tweaked from {!CodeGen.ml}. ----- *)
 let is_foreign_array (pt: Ast.parameter_type) =
   match pt with
-      Ast.PTVal _     -> false
-    | Ast.PTPtr(t, a) ->
-        match t with
-            Ast.Foreign _ -> a.Ast.pa_isary
-          | _             -> false
+    Ast.PTVal _ -> false
+  | Ast.PTPtr(t, a) ->
+    match t with
+      Ast.Foreign _ -> a.Ast.pa_isary
+    | _ -> false
+
+(** Get the array declaration from a list of array dimensions. Empty
+    [ns] indicates the corresponding declarator is a simple identifier.
+    Element of value -1 means that user does not specify the dimension
+    size. *)
 let get_array_dims (ns: int list) =
-  (* Get the array declaration from a list of array dimensions.
-   * Empty `ns' indicates the corresponding declarator is a simple identifier.
-   * Element of value -1 means that user does not specify the dimension size.
-   *)
   let get_dim n = if n = -1 then "[]" else sprintf "[%d]" n
   in
-    if ns = [] then ""
-    else List.fold_left (fun acc n -> acc ^ get_dim n) "" ns
+  if ns = [] then ""
+  else List.fold_left (fun acc n -> acc ^ get_dim n) "" ns
 
 let get_typed_declr_str (ty: Ast.atype) (declr: Ast.declarator) =
   let tystr = Ast.get_tystr  ty in
   let dmstr = get_array_dims declr.Ast.array_dims in
-    sprintf "%s %s%s" tystr declr.Ast.identifier dmstr
+  sprintf "%s %s%s" tystr declr.Ast.identifier dmstr
 
-(* Check whether given parameter is `const' specified. *)
+(** Check whether given parameter [pt] is [const] specified. *)
```

Comment:
What does putting certain words within [ ] do?

Reply:
It's OCamldoc syntax for pre-formatted code, similar to backticks in Markdown, or backtick/single-quote in Lisp. This way we can later generate some formatted/rendered docs out of this.